test-data/unit/check-newsemanal.test

```diff
@@ -2756,3 +2756,14 @@ def g() -> None:
     x = f()  # E: Need type annotation for 'x'
     y = x
     reveal_type(y)  # N: Revealed type is '__main__.G[Any]'
+
+[case testNewAnalyzerRedefinedNonlocal]
+import typing
+
+def main():
+    bar = []  # type: typing.List[int]
+
+    def foo():
```

Comment:
Is there any difference in behavior if these functions are annotated (i.e. checked)? Maybe we should have two tests (with and without annotations)?